cmd/agent/service_windows.go

```diff
@@ -74,20 +80,23 @@ loop:
 			case svc.Interrogate:
 				changes <- c.CurrentStatus
 			case svc.Stop, svc.Shutdown:
-				ep.Stop()
```

Comment:
Hmm, doesn't this mean there's no way to stop the service? Finding a way to tie this in cleanly would be a PITA but IMO for now we can give `ep.Start` a context that we can cancel whenever we need it to shut down.

Reply:
When you say service do you mean the Windows service or entrypoint? The windows service will stop when this returns and the entrypoint will normally stop on line 98. Though there is a possible race condition.

Reply:
Oh, I see, right. Yeah, this is probably fine as-is. Never mind :)